ble/BLE.h

```diff
@@ -1150,6 +1150,11 @@ class BLE
         gap().onConnection(connectionCallback);
     }
 
+    template<typename T>
```

Comment:
Hi, can we not do this for BLE.h (because that's deprecated space)? let's limit it to Gap.h?
thanks.